src/Snippets/Docs/Migration.Execute.cs

```diff
@@ -0,0 +1,100 @@
+using Polly.Timeout;
+
+namespace Snippets.Docs;
+
+internal static partial class Migration
+{
+    private static int Method() => 3;
+    private static Task<int> MethodAsync() => Task.FromResult(3);
+    public static async Task SafeExecute_V7()
+    {
+        #region migration-execute-v7
+        // Synchronous execution
+        ISyncPolicy<int> syncPolicy = Policy.Timeout<int>(TimeSpan.FromSeconds(1));
+        PolicyResult<int> policyResult = syncPolicy.ExecuteAndCapture(Method);
+
+        // Asynchronous execution
+        IAsyncPolicy<int> asyncPolicy = Policy.TimeoutAsync<int>(TimeSpan.FromSeconds(1));
+        PolicyResult<int> asyncPolicyResult = await asyncPolicy.ExecuteAndCaptureAsync(MethodAsync);
+
+        // Assess policy result
+        if (policyResult.Outcome == OutcomeType.Successful)
+        {
+            var result = policyResult.Result;
+            // process result
+        }
+        else
+        {
+            var exception = policyResult.FinalException;
+            FaultType failtType = policyResult.FaultType!.Value;
+            ExceptionType exceptionType = policyResult.ExceptionType!.Value;
+            // process failure
+        }
+
+        // Access context
+        IAsyncPolicy<int> asyncPolicyWithContext = Policy.TimeoutAsync<int>(TimeSpan.FromSeconds(10),
+            onTimeoutAsync: (ctx, ts, task) =>
+            {
+                ctx["context_key"] = "context_value";
+                return Task.CompletedTask;
+            });
+
+        asyncPolicyResult = await asyncPolicyWithContext.ExecuteAndCaptureAsync((ctx) => MethodAsync(), new Context());
+        var ctxValue = asyncPolicyResult.Context.GetValueOrDefault("context_key");
+        #endregion
+    }
+
+    public static async Task SafeExecute_V8()
+    {
+        #region migration-execute-v8
+        ResiliencePipeline<int> pipeline = new ResiliencePipelineBuilder<int>()
+            .AddTimeout(TimeSpan.FromSeconds(1))
+            .Build();
+
+        // Synchronous execution
+        // Polly v8 does not provide an API to synchronously execute and capture the outcome of a pipeline
+
+        // Asynchronous execution
+        var context = ResilienceContextPool.Shared.Get();
+        Outcome<int> pipelineResult = await pipeline.ExecuteOutcomeAsync(
```

Comment:
@martincostello , @peter-csala 

Wondering whether it makes sense to add more `ExecuteOutcome` convenience overloads so we do not force passing `ResilienceContext` and `state`?

Or maybe wait for feedback until someone really needs these?

Reply:
Should be easy enough for people to just use a discard if they don't need them?

Reply:
IMHO it would definitely ease the usage of the `ExecuteOutcomeAsync` method.

Reply:
One reason why there is only a single method is that I considered this advanced API that won't be used that much. But maybe I am wrong and it will be commonly used. 

In that case having some convenience overloads makes sense (don't force  people to use `ResilienceContextPool.Shared.Get()`). 

But maybe let's just have "wait and see" approach? Adding new APIs won't be a breaking change anyway.